### PR TITLE
Slow walk

### PR DESCRIPTION
**Pull request by notgoodusename**
redone

### CODE REVIEW

**Comment by danielkrupinski** [comment]
What's the purpose of this feature?

**Comment by notgoodusename** [comment]
the purpose is to walk slow enogh to make your self accurate at the same time maintaining a good fake

**Comment by danielkrupinski** [comment]
I think it should calculate slow walk speed from active weapon data to work for every weapon automatically.

File: Osiris/Hacks/Misc.cpp [hunk 1]
```diff
@@ -10,7 +10,28 @@
...
+    if (amount <= 0) {
+        return;
+    }
+    if (GetAsyncKeyState(config.misc.slowwalkkey) && config.misc.slowwalk) {
+        auto localPlayer = interfaces.entityList->getEntity(interfaces.engine->getLocalPlayer());
+        Vector velocity = localPlayer->velocity();
+        static Vector direction = { 0,0,0 };
+        float speed = velocity.length();
+        direction.y = cmd->viewangles.y - direction.y;
+        Vector forward;
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp]: uninitialized

File: Osiris/Hacks/Misc.cpp [hunk 2]
```diff
@@ -10,7 +10,28 @@
...
+ void Misc::slowwalk(UserCmd* cmd) noexcept
+{
+    int amount = config.misc.slowwalkammount;
+    if (amount <= 0) {
+        return;
+    }
+    if (GetAsyncKeyState(config.misc.slowwalkkey) && config.misc.slowwalk) {
+        auto localPlayer = interfaces.entityList->getEntity(interfaces.engine->getLocalPlayer());
+        Vector velocity = localPlayer->velocity();
+        static Vector direction = { 0,0,0 };
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp]: unused variable

**Comment by notgoodusename** [comment]
one question tho it osiris doesnt have max speed

**Comment by danielkrupinski** [comment]
Now it has 4ca1f15

**Comment by notgoodusename** [comment]
I cant edit the repository because is gone tho what i have now is this
` void Misc::slowwalk(UserCmd* cmd) noexcept
{
    int amount = config.misc.slowwalkammount;
    if (amount <= 0) {
        return;
    }
    if (GetAsyncKeyState(config.misc.slowwalkkey) && config.misc.slowwalk) {

        auto localPlayer = interfaces.entityList->getEntity(interfaces.engine->getLocalPlayer());
        float speed = localPlayer->velocity().length();
        auto weapon = localPlayer->getActiveWeapon();
        static Vector forward = cmd->viewangles;
        Vector source = forward * -speed;
        if (speed >= (weapon->getWeaponData()->maxSpeed)*amount)
        {
            cmd->forwardmove = source.x;
            cmd->sidemove = source.y;

        }
    }
}`

**Pull request closed without merging**